src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/LocalAppContextSwitches.cs

```diff
@@ -270,5 +272,29 @@ public static bool LegacyVarTimeZeroScaleBehaviour
                 return s_legacyVarTimeZeroScaleBehaviour == Tristate.True;
             }
         }
+
+        /// <summary>
+        /// When set to true, the connection pool will use the new V2 connection pool implementation.
+        /// When set to false, the connection pool will use the legacy V1 implementation.
+        /// This app context switch defaults to 'false'.
+        /// </summary>
+        public static bool UseConnectionPoolV2
+        {
+            get
+            {
+                if (s_useConnectionPoolV2 == Tristate.NotInitialized)
+                {
+                    if (AppContext.TryGetSwitch(UseConnectionPoolV2String, out bool returnedValue) && returnedValue)
```

Comment:
I would consider turning this into a helper method or a ternary statement, since we probably use this pattern a lot and it's kinda verbose for what it does.